### ISSUE TITLE
`get_frontal_face_detector` vs DNN MMOD Face Detector

### ISSUE THREAD

**Issue by bkj** [davisking/dlib#447]
The blog entry below shows performance of dlib's DNN MMOD face detector compared against Viola Jones and Faster-RCNN.  Are there metrics anywhere for the performance of the face detector loaded by the (Python) `dlib.get_frontal_face_detector`?  The DNN MMOD code doesn't seem to have a Python API yet, so I was wondering how big the benefit of switching would be.

http://blog.dlib.net/2016/10/easily-create-high-quality-object.html

**Comment by davisking** [comment]
The MMOD paper contains some FDDB evaluations of the HOG style detector.  Although, the HOG detector given by `dlib.get_frontal_face_detector` is actually a combination of 5 HOG detectors and so is somewhat more accurate than the single HOG evaluation in the MMOD paper.  

But anyway, I have evaluated it on FDDB but don't have any results on hand anymore.  Although that blog post you mentioned should give you a pretty good idea what the difference is.  It should also be pointed out that FDDB is somewhat of a dated evaluation with a pretty huge dataset bias, as also discussed in the blog post you mentioned.

**Comment by mrgloom** [comment]
How Viola-Jones detector was evaluated? As I understand fddb benchmark requires detector to produce `score` of detection, for example opencv Viola-Jones implementation doesn't produce the score, so how it was calculated? you were using some custom implementation?

**Comment by davisking** [comment]
http://vis-www.cs.umass.edu/fddb/results.html

**Comment by abhisheksgumadi** [comment]
Hi @davisking thanks for the fantastic work in DLIb. I was reading your blog where you said the CNN is trained on just 4 images. Is that true? Does that mean it used just 4 images ! to train its end-end CNN with MMOD loss function? I am wondering why such a small data set of just 4 images? Thanks.

**Comment by davisking** [comment]
The example program trains with just 4 images. The pretrained CNN model file that you can download from dlib.net is trained with thousands.  This is all discussed in the documentation.
